spec/models/asso_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "spec_helper"
+
+RSpec.describe Asso, type: :model do
+
+  describe "#create_new_asso_and_relation" do
+    let!(:factory1) { create :factory }
+    let!(:asso1) { create :asso, name: "asso1", factory_id: factory1.id }
+    let!(:parent1) { create :factory, name: "parent1" }
+    let!(:asso_relation1) { create(:asso_relation, asso_id: asso1.id, factory_id: parent1.id) }
+    it "same asso don't exist case" do
+      expect{ Asso.create_new_asso_and_relation({ name: "asso2", factory_name: factory1.name, traits: []}, parent1) }.to change { Asso.all.size }.from(1).to(2)
```

Comment:
Style/SpaceBeforeBlockBraces: Space missing to the left of {.<br>Style/SpaceInsideHashLiteralBraces: Space inside } missing.